### PR TITLE
Add a `plutus` executable

### PR DESCRIPTION
**Pull request by michaelpj**
This adds a `plutus` executable with two subcommands: `typecheck` and `evaluate`. The input can be passed either as a file or as stdin.

I've also removed `run-ck`, since `plutus evaluate` subsumes it, but I could leave it if we wanted.

### CODE REVIEW

File: language-plutus-core/plutus-exe/Main.hs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+-- | Parse a program and run it using the CK machine.
+parseRunCk :: BSL.ByteString -> Either (PC.ParseError PC.AlexPosn) CK.CkEvalResult
+parseRunCk = fmap (CK.runCk . void) . PC.parseScoped
+
+-- | Parse a program and typecheck it.
+parseTypecheck :: BSL.ByteString -> Either (PC.Error PC.AlexPosn) (PC.Type PC.TyNameWithKind ())
+parseTypecheck bs = runExcept $ PC.runQuoteT $ do
+    parsed <- PC.parseProgram bs
+    annotated <- PC.annotateProgram parsed
```
[the remark this answers is not in the document]
Comment (michaelpj) [reply language-plutus-core/plutus-exe/Main.hs]: Sure - but I'll do that separately.

**Comment by michaelpj** [comment]
Docs updated.

**Comment by effectfully** [comment]
Should we call it `plutus` rather than `plutus-core`? We'll have `plutus-tx` later.

**Comment by michaelpj** [comment]
I had been thinking that `plutus-tx` isn't going to have an executable, since you can only really use it via Haskell. I'd also like a relatively short name - how about `plc`?

**Comment by effectfully** [comment]
`plc` is fine to me.

**Comment by michaelpj** [comment]
I've renamed it (and updated the docs again!).

**Pull request merged**